[DOCS] Updates 7.0 upgrade guide (#34121)
elastic/kibana: docs/setup/upgrade.asciidoc

* Backup your data using the {es} {ref}/modules-snapshots.html[snapshots] feature.
  You **cannot roll back** to an earlier version unless you have a backup of
  your data.
* If you are using custom plugins, check that a compatible version is
  available.
===========================================

Depending on which version of {kib} you're upgrading from, the upgrade process to 7.0
will vary. 

The recommended path is to upgrade to 6.7 before upgrading to 7.0. 
This makes it easier to identify the changes you need to make to upgrade and enables 
you to perform a rolling upgrade with no downtime. 

Please see 
*{stack-ref}/upgrading-elastic-stack.html[Upgrading the Elastic Stack]* for a 
comprehensive overview of the upgrade process.

[float]
==== Upgrading from 6.6 or earlier
{es} can read indices created in the previous major version. If you have indices 
created in 5.x or before, you must reindex or delete them before upgrading to 7.0.0.
See {stack-ref}/upgrading-elastic-stack.html#oss-stack-upgrade[Upgrading from 6.6 or earlier] 
for more information.

Once your reindex is complete, you can follow the <<upgrade-standard, Standard upgrade>> 

> instructions to do a manual upgrade. The recommended path is to upgrade to 6.7 before upgrading to 7.0.

Reviewer: I'm not sure what a "manual upgrade" is.  It implies there's an automatic upgrade.  If there's no such thing, I'd suggest dropping that phrase:

```suggestion
instructions. The recommended path is to upgrade to 6.7 before upgrading to 7.0.
```

*Note:* Saved searches, visualizations, and dashboards created in {kib} 6.x
can be generally imported into 7.x.

[float]
==== Upgrading from 6.7
6.7 includes an https://www.elastic.co/guide/en/kibana/6.7/upgrade-assistant.html[Upgrade Assistant] 
to help you prepare for your upgrade to 7.0. To access the assistant, go to *Management > 7.0 Upgrade Assistant*. 

> 

Reviewer: FYI: The last step in the Upgrade Assistant is a referral to these upgrading docs:
![image](https://user-images.githubusercontent.com/26471269/55244570-a7301500-51fe-11e9-9bd0-9abe2b52a966.png)
That is to say, it does not do the upgrade for you.  Therefore I think the "You can also..." sentence incorrectly implies that you either use the Upgrade Assistant OR do the upgrade manually.

Author: Good point, thanks

> You can also use the <<upgrade-standard, Standard upgrade>> instructions to do a manual upgrade to 7.0.

Reviewer: Ditto about dropping "manual upgrade":  

```suggestion
You can also use the <<upgrade-standard, Standard upgrade>> instructions to upgrade to 7.0.
```

> You can also use the <<upgrade-standard, Standard upgrade>> instructions to do a manual upgrade to 7.0.

Reviewer: ```suggestion
After you have addressed any issues that were identified by the Upgrade Assistant, <<upgrade-standard,upgrade to 7.0>>.
```

* Backup your data using the {es} {ref}/modules-snapshots.html[snapshots] feature.
  You **cannot roll back** to an earlier version unless you have a backup of
  your data.
* If you are using custom plugins, check that a compatible version is
  available.
===========================================

Depending on which version of {kib} you're upgrading from, the upgrade process to 7.0
will vary. 

The recommended path is to upgrade to 6.7 before upgrading to 7.0. 
This makes it easier to identify the changes you need to make to upgrade and enables 
you to perform a rolling upgrade with no downtime. 

Please see 
*{stack-ref}/upgrading-elastic-stack.html[Upgrading the Elastic Stack]* for a 
comprehensive overview of the upgrade process.

[float]
==== Upgrading from 6.6 or earlier
{es} can read indices created in the previous major version. If you have indices 
created in 5.x or before, you must reindex or delete them before upgrading to 7.0.0.
See {stack-ref}/upgrading-elastic-stack.html#oss-stack-upgrade[Upgrading from 6.6 or earlier] 
for more information.

Once your reindex is complete, you can follow the <<upgrade-standard, Standard upgrade>> 

> instructions. The recommended path is to upgrade to 6.7 before upgrading to 7.0.

Reviewer 2: I think it would be good to call out here that this manual reindex process is not necessary if they upgrade to 6.7 and use the Upgrade Assistant instead.

Author: @joshdover Wouldn't someone on a much earlier version (say, 4.x) need to go through a reindex to be able to get to 6.7, before they could use the Upgrade Assistant to prepare for the upgrade to 7.0?

Reviewer 2: They will, but they also will have to through 5.6 first as well. If you're on 2.x for instance, the process would be
- Reindex 1.x index
- Upgrade to 4.x
- Reindex every 2.x index
- Upgrade to 5.6
- Reindex every 4.x index
- Upgrade to 6.7
- Reindex every 5.x index

I think the linked stack doc makes it pretty clear that upgrading directly from 4.x isn't supported:
> Elasticsearch supports rolling upgrades between minor versions, from Elasticsearch 5.6 to 6.7, and from 6.7 to 7.0.0-rc1.

In that case, the customer should be first upgrading and following the docs for 4.x to 5.6. Once they're on 5.6, they can use the upgrade assistant in 5.6 to reindex without having to do this manually. I'm not sure the best way to convey this in the documentation besides linking to the docs for upgrading to 5.6 first?

Author: Right, okay - so maybe noting with a link that an Upgrade Assistant is available in 5.6? As you note, I want to avoid detailing every upgrade scenario, and assume that people will reference the upgrade docs from previous versions (if that is their starting point).

Author: Thanks - pushed some updates
